server/routes/volunteers.js

```diff
@@ -8,17 +8,44 @@ const uuid = require('uuid/v1');
 
 // Get all volunteers for department
 router.get('/departments/:departmentId/volunteers', co.wrap(function* (req, res) {
-  const departmentId = req.params.departmentId;
-  const department = yield Department.findOne({_id: departmentId, deleted: false});
-  if (_.isEmpty(department)) return res.status(404).json({error: `Department ${departmentId} does not exist`});
+    const departmentId = req.params.departmentId;
+    const department = yield Department.findOne({_id: departmentId, deleted: false});
+    if (_.isEmpty(department)) return res.status(404).json({error: `Department ${departmentId} does not exist`});
+
+    // TODO: check permission
+
+    const departmentVolunteers = yield Volunteer.find({departmentId: departmentId, deleted: false});
+    let departmentVolunteersUserIds = departmentVolunteers.map(volunteer => volunteer.userId);
+    if (departmentVolunteersUserIds && departmentVolunteersUserIds.length > 0) {
+        const departmentVolunteersOtherDeparments = yield Volunteer
+            .find({
+                userId: {$in: departmentVolunteersUserIds},
+                deleted: false
+            });
+
+        let otherDepartmentIds = departmentVolunteersOtherDeparments.map(volunteer => volunteer.departmentId);
+        const otherDepartmentObjects = yield Department.find({_id: {$in: otherDepartmentIds}, deleted: false});
```

Comment:
No need to return the entire department, id its enough and we can complete the names on the client. 
Also its a lot of filtering inside a for each, maybe its better to create a dictionary of volunteers by id for both queries (departmentVolunteersOtherDeparments, departmentVolunteers) and join them.